### PR TITLE
Fix version 2.1 release note regarding TF_DETERMINISTIC_OPS

### PR DESCRIPTION
**Pull request by duncanriach**
**This current pull request is intentionally aimed at the r2.1 branch.**

This current pull request is a follow-up to [PR 34887](https://github.com/tensorflow/tensorflow/pull/34887). Unfortunately, I didn't review the rendered markdown for those changes before submitting the PR.

This PR:
  * Changes the indent of the note about `TF_DETERMINISTIC_OPS` so that it doesn't appear to be related to TensorRT.
  * Escapes the use of \* twice in the markdown, which was previously interpreted as intending to signal emphasis.
  * Adds some improvements to clarity.

Requesting review by @sanjoy, who reviewed the original PR.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ TensorFlow 2.1 will be the last TF release supporting Python 2. Python 2 support
...
-  * Environment variable `TF_DETERMINISTIC_OPS` added. When set to "true" or "1", this environment variable makes `tf.nn.bias_add` operate deterministically (i.e. reproducibly) when XLA JIT compilation is *not* enabled. It also makes cuDNN convolution and max-pooling operate deterministically. This makes Keras Conv*D and MaxPool*D layers operate deterministically in both the forward and backward directions when running on a CUDA-enabled GPU.
+* Environment variable `TF_DETERMINISTIC_OPS` has been added. When set to "true" or "1", this environment variable makes `tf.nn.bias_add` operate deterministically (i.e. reproducibly), but currently only when XLA JIT compilation is *not* enabled. `TF_DETERMINISTIC_OPS` also makes cuDNN convolution and max-pooling operate deterministically. This makes Keras Conv\*D and MaxPool\*D layers operate deterministically in both the forward and backward directions when running on a CUDA-enabled GPU.
```
Comment (sanjoy) [inline RELEASE.md]: Up to you, but I'd also wordsmith "`TF_DETERMINISTIC_OPS` also makes cuDNN ..." to "Setting `TF_DETERMINISTIC_OPS` to "true" or "1" also makes cuDNN ..." or say "It also makes cuDNN ..."

**Review by sanjoy — DISMISSED** [review]
Thanks!  And sorry for not noticing earlier!

File: RELEASE.md (hunk 1 above)
Comment (duncanriach) [reply RELEASE.md]: No problem, Sanjoy. You were focused on the technical content in the previous PR. Great suggestion regarding the wordsmithery. I'l make that change now ...

**Review by sanjoy — APPROVED** [review]
lgtm

**Comment by duncanriach** [comment]
This PR has been ready to merge for 9 days. Please will someone merge it?

**Comment by mihaimaruseac** [comment]
Apologies, we didn't merge any PRs on the branch this week as we were debugging several segfaults caused by scipy/scipy#11237

We will start merging PRs in a while, when we finish this debugging process

**Comment by goldiegadde** [comment]
@duncanriach sorry for the delay in merging the PR's. we have been debugging seg fault issues with scipy and hence the delay. Should be merged in soon.

**Comment by duncanriach** [comment]
@goldiegadde, thanks for the update and thanks for the merge.

**Pull request merged**